src/Storages/ObjectStorage/DataLakes/Iceberg/Mutations.cpp

```diff
@@ -60,8 +64,14 @@ using DeleteFileStatisticsByPartitionKey = std::unordered_map<ChunkPartitioner::
 
 struct DeleteFileWriteResultWithStats
 {
-    DeleteFileWriteResultByPartitionKey data_file;
-    DeleteFileStatisticsByPartitionKey statistic;
+    DeleteFileWriteResultByPartitionKey delete_file;
+    DeleteFileStatisticsByPartitionKey delete_statistic;
+};
+
+struct WriteDataFilesResult
+{
+    DeleteFileWriteResultWithStats delete_file;
+    std::optional<DeleteFileWriteResultWithStats> data_file;
```

Comment:
It looks confusing that  `data_file` also has type is `DeleteFileWriteResultWithStats`, while it is not related to delete (currently it is uses as update data file result and stats). I'd rename (for example to something  common like  `DataFileWriteResultWithStats`?), otherwise it makes the code more difficult to read and understand